compiler/rustc_typeck/src/astconv/mod.rs

```diff
@@ -1338,7 +1337,7 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
         }
 
         // Use a `BTreeSet` to keep output in a more consistent order.
-        let mut associated_types: FxHashMap<Span, BTreeSet<DefId>> = FxHashMap::default();
+        let mut associated_types: FxHashMap<Span, FxHashSet<DefId>> = FxHashMap::default();
```

Comment:
Do we still rely on the iteration order?

Reply:
Yes: https://github.com/rust-lang/rust/blob/181e91567c9f347e055b33b1d7e9894f769aafe3/compiler/rustc_typeck/src/astconv/mod.rs#L1403